examples/server/tests/features/environment.py

```diff
@@ -29,31 +29,43 @@ def after_scenario(context, scenario):
                     for line in f:
                         print(line)
         if not is_server_listening(context.server_fqdn, context.server_port):
-            print("\x1b[33;101mERROR: Server stopped listening\x1b[0m")
+            print("\x1b[33;101mERROR: Server stopped listening\x1b[0m\n")
 
     if not pid_exists(context.server_process.pid):
         assert False, f"Server not running pid={context.server_process.pid} ..."
 
-    print(f"stopping server pid={context.server_process.pid} ...")
-    context.server_process.kill()
+    kill_server(context)
+
     # Wait few for socket to free up
     time.sleep(0.05)
 
     attempts = 0
     while is_server_listening(context.server_fqdn, context.server_port):
-        print(f"stopping server pid={context.server_process.pid} ...")
-        os.kill(context.server_process.pid, SIGKILL)
+        context.server_process.kill()
         time.sleep(0.1)
         attempts += 1
         if attempts > 5:
-            print(f"Server dangling exits, killing all {context.server_path} ...")
-            process = subprocess.run(['killall', '-9', context.server_path],
-                                     stderr=subprocess.PIPE,
-                                     universal_newlines=True)
+            if os.name == 'nt':
+                print(f"Server dangling exits, task killing force {context.server_process.pid} ...\n")
+                process = subprocess.run(['taskkill', '/F', '/pid', str(context.server_process.pid)],
+                                         stderr=subprocess.PIPE)
+            else:
+                print(f"Server dangling exits, killing all {context.server_path} ...\n")
+                process = subprocess.run(['killall', '-9', context.server_path],
+                                         stderr=subprocess.PIPE)
```

Comment:
I know that at least from the Unix side of things, you should *never* have to shell out to killall to kill a child process. You can kill the entire process group with `os.kill(-pid, signal.SIGKILL)` - nothing more than that should be necessary.

Reply:
Agreed, but I added it at the very beginning because kill() was actually not working. If we arrive here, it means several calls to kill()/kill(sigkill) have been made without success because the server is still listening. 
I will remove it, we will see. Anyway, I also want to test the grateful shutdown, so let's enjoy this pr for that